File: debrujin/python/src/1.0.1.py
def debrujin(alpha, n):
  base = len(alpha)
  highestValue = base ** n - 1
  endStringLength = highestValue + n
  string = alpha[0] * n + alpha[1]
  alphaLocation = 0
  while len(string) < endStringLength:
    chunk = string[1 - n:]
    beta = alpha[alphaLocation]
    if string.find(chunk + beta) + 1:
      alphaLocation += 1
      if alphaLocation >= base:
        lastChar = string[-1:]
        lastCharPos = alpha.find(lastChar)
        alphaLocation = lastCharPos + 1
        string = string[:-1]
    else:
      string += beta
      alphaLocation = 0
  return string

File: debrujin/javascript/src/1.0.1.js
function debrujin(alpha, n) {
  const base = alpha.length;

  let string = alpha[0].repeat(n);
  let alphaLocation = 0;
  for (;string.length < parseInt(String(base - 1).repeat(n), base) + n;) {
    const beta = alpha[alphaLocation];
    if (string.match(string.slice(string.length - n + 1, string.length) + beta)) {
      if (++alphaLocation >= base) {
        alphaLocation = alpha.indexOf(string[string.length - 1]) + 1;
        string = string.replace(/.$/, '');
      }
    } else {
      string += beta;
      alphaLocation = 0;
    }
  }
  return string;
}